server/bin/pbench-restore-and-unpack-tarballs.py

```diff
@@ -0,0 +1,243 @@
+                    flush=True,
+                )
+                continue
+
+            # Create controller directory if it doesn't exist
+            ctrl_p = archive_p / ctrl
+            if not options.dry_run:
+                try:
+                    ctrl_p.mkdir()
+                except FileExistsError:
+                    pass
+                else:
+                    print(f"\nCreated controller directory, '{ctrl_p}'", flush=True)
+
+            # Copy tar ball to archive tree, along with its .md5 ...
+            cp_cmd = f"cp -a {tb} {tb}.md5 {ctrl_p}/"
+            print(f"\n{cp_cmd}", flush=True)
+            if not options.dry_run:
+                cp = subprocess.run(cp_cmd, shell=True, stderr=subprocess.STDOUT)
+                if cp.returncode != 0:
+                    print(
+                        f"FAILURE: cp command: '{cp_cmd}': {cp.returncode}, {cp.stdout!r}",
+                        file=sys.stderr,
+                    )
+                    sys.exit(1)
+            # ... and run md5sum check against it
+            md5sum_cmd = f"md5sum --check {a_tb}.md5"
+            print(md5sum_cmd, flush=True)
+            if not options.dry_run:
+                cp = subprocess.run(md5sum_cmd, shell=True, cwd=str(ctrl_p))
+                if cp.returncode != 0:
+                    print(
+                        f"FAILURE: md5sum command: '{md5sum_cmd}': {cp.returncode}, {cp.stdout!r}",
+                        file=sys.stderr,
+                    )
+                    sys.exit(1)
+
+            # Create the symlink recording the tar ball is already backed up
+            backed_up_dir = ctrl_p / "BACKED-UP"
+            if not options.dry_run:
+                backed_up_dir.mkdir(exist_ok=True)
+            backed_up = backed_up_dir / tb.name
+            print(f"ln -s {a_tb} {backed_up}", flush=True)
+            if not options.dry_run:
+                backed_up.symlink_to(a_tb)
+
+            # Create the symlink requesting the tar ball be unpacked
+            to_unpack_dir = ctrl_p / "TO-RE-UNPACK"
+            if not options.dry_run:
+                to_unpack_dir.mkdir(exist_ok=True)
+            to_unpack = to_unpack_dir / tb.name
+            print(f"ln -s {a_tb} {to_unpack}", flush=True)
+            if not options.dry_run:
+                to_unpack.symlink_to(a_tb)
+
+            tbs_restored += 1
+
+            print(
+                f"\nRestored {tbs_restored:d} ({tbs_existing:d} existing) of ({tbs_cnt:d}): {a_tb}\n",
+                flush=True,
+            )
+        except Exception as exc:
+            print(
+                f"Exception while processing ({tb_dt!r}, {ctrl!r}, {tb!r}): {exc}",
+                file=sys.stderr,
+            )
+            raise
+
+    end = pbench._time()
+    if options.dry_run:
+        print(
+            f"Restored {tbs_restored:d} tar balls from backup volume,"
+            f" {backup} (ended at: {end}, elapsed ({end - start:0.2f})"
+        )
```

Comment:
Why is this printed only for dry runs?  (And, why bother timing it?...won't t take _much_ longer if we actually do the work?).  Why note only the restored files and not the existing count or the total count?

And, given that this is only for dry runs, again, it would be better if the message made clear that it didn't actually do anything.